src/encoded/commands/purge_item_type.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import argparse
+import logging
+import structlog
+import transaction
+from pyramid.paster import get_app
+
+from dcicutils.env_utils import is_stg_or_prd_env
+from snovault import STORAGE
+from snovault.elasticsearch.indexer_utils import get_uuids_for_types
+from .. import configure_dbsession
+
+
+logger = structlog.getLogger(__name__)
+EPILOG = __doc__
+
+
+def purge_item_type_from_storage(app, item_types, prod=False):
+    """
+    Purges all items with the given item_types from our storage. This function could partially purge
+    an item type if an error is encountered. Note that this will work no matter what resources are backing
+    'PickStorage'.
+
+    IMPORTANT: If an error occurs the DB transaction is rolled back, but the ES deletions will persist.
```

Comment:
If you were to commit on every iteration, then the most you'd be out is 1 item when you got an error and perhaps you could hold the old value of it and resubmit it in that case, or else just recompute it (though that might lead to subtle differences).